ui/components/app/modals/account-details-modal/account-details-modal.component.js

```diff
@@ -2,11 +2,18 @@ import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { getAccountLink } from '@metamask/etherscan-link';
 
+import { ethers } from 'ethers';
 import AccountModalContainer from '../account-modal-container';
 import QrView from '../../../ui/qr-code';
 import EditableLabel from '../../../ui/editable-label';
 import Button from '../../../ui/button';
 
+import ENS from '../../../../../app/scripts/controllers/ens/ens';
```

Comment:
[req] unfortunately, ENS is a controller that we should only utilize in the background. Fortunately, though there is an action for this already! tryReverseResolveAddress from the redux actions. For an example of usage, see the transaction-list-item-details.component and .container files.

Reply:
Thanks for your advice. I just changed the component to a functional type.
And instead of using tryReverseResolveAddress in account-details-modal.component.js, I use ensResolutionByAddress to get ENS in account-details-modal.container.js. 
Is it okay to do this?

```
account-details-modal.container.js

const mapStateToProps = (state) => {
  const { metamask } = state;
  const { ensResolutionsByAddress } = metamask;
  const selectedIdentity = getSelectedIdentity(state);
  let ensName;
  if (selectedIdentity.address) {
    const address = toChecksumHexAddress(selectedIdentity.address);
    ensName = ensResolutionsByAddress[address] || '';
  }
  return {
    chainId: getCurrentChainId(state),
    selectedIdentity,
    keyrings: state.metamask.keyrings,
    rpcPrefs: getRpcPrefsForCurrentProvider(state),
    ensName,
  };
};
```

Reply:
Oh, I need both. I will upload code soon:)